### PR TITLE
Full project name

### PR DESCRIPTION
**Pull request by mwinter69**
When folders are installed the projectName alone might not be unique anymore. So we should also expose the fullProjectName and rootFullProjectName.

### CODE REVIEW

File: src/test/java/jenkins/plugins/logstash/persistence/BuildDataTest.java [hunk 1]
```diff
@@ -198,37 +183,14 @@ public void constructorSuccessBuiltOnSlave() throws Exception {
...
-    verify(mockDate).getTime();
+    verifyMocks();
+    verifyTestResultActions();
   }
 
   @Test
   public void constructorSuccessTestFailures() throws Exception {
     TestResult mockTestResult = Mockito.mock(hudson.tasks.test.TestResult.class);
-    when(mockTestResult.getSafeName()).thenReturn("Mock Test");
+    //when(mockTestResult.getSafeName()).thenReturn("Mock Test");
```
Comment (jakub-bochenski) [inline src/test/java/jenkins/plugins/logstash/persistence/BuildDataTest.java]: Please don't leave commented out code

If this method is no longer called please just remove it

Comment (mwinter69) [reply src/test/java/jenkins/plugins/logstash/persistence/BuildDataTest.java]: removed

**Comment by jakub-bochenski** [comment]
@mwinter69 sorry, but I still can't find any new test cases in the diff

**Comment by mwinter69** [comment]
I reworked the toJsonSuccess test in BuildDataTest. It now will be passed a mocked Build. And then the resulting json string is verified. This basically tests all the things where information from Build/project is read including the fullProjectName.

File: src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 2]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (jakub-bochenski) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: Is this really needed? I don't think the code is mocking any final/static methods now?

**Comment by jakub-bochenski** [comment]
@mwinter69 yes, but I'd like to have a separate test case for each new feature.
Some call it "One assertion per test" rule,  e.g.  28b4ecd85013278a9b47564d06867f8ff5646b3d

I know the current test code is of low quality. I don't have the spare time to rewrite it from scratch. That's why I'm asking for incremental improvements on PRs.

Still if you don't want to do this I won't insist. I have one more question and we can merge this.

File: src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 2 above)
Comment (mwinter69) [reply src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker]: It is needed. The method AbstractItem.getFullName() is final

**Pull request merged**